HTTP 500 for oauth2/userinfo endpoint when invalid access token is used.

There is an error when invoked the oauth2/userinfo endpoint with the expired or invalid access token. This is happening when IS Analytics is enabled. The usual response is 401. Please find the below stack trace from IS.

`[2019-11-27 21:49:57,867] ERROR - AbstractFaultChainInitiatorObserver An unexpected error occurred during error handling. No further error processing will occur.
org.apache.cxf.interceptor.Fault: Invalid access token
                at org.apache.cxf.service.invoker.AbstractInvoker.createFault(AbstractInvoker.java:162)
                at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:128)
                at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:193)
                at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:103)
                at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59)
                at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96)
                at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
                at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121)
                at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:267)
                at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234)
                at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208)
                at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160)
                at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:216)
                at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:301)
                at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doGet(AbstractHTTPServlet.java:225)
                at javax.servlet.http.HttpServlet.service(HttpServlet.java:624)
                at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:276)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
                at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
                at com.thetransactioncompany.cors.CORSFilter.doFilter(CORSFilter.java:168)
                at com.thetransactioncompany.cors.CORSFilter.doFilter(CORSFilter.java:233)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
                at org.apache.catalina.filters.HttpHeaderSecurityFilter.doFilter(HttpHeaderSecurityFilter.java:126)
                at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
                at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
                at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:219)
                at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:110)
                at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:494)
                at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:169)
                at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:104)
                at org.wso2.carbon.tomcat.ext.valves.CompositeValve.continueInvocation(CompositeValve.java:99)
                at org.wso2.carbon.tomcat.ext.valves.CarbonTomcatValve$1.invoke(CarbonTomcatValve.java:47)
                at org.wso2.carbon.webapp.mgt.TenantLazyLoaderValve.invoke(TenantLazyLoaderValve.java:57)
                at org.wso2.carbon.tomcat.ext.valves.TomcatValveContainer.invokeValves(TomcatValveContainer.java:47)
                at org.wso2.carbon.tomcat.ext.valves.CompositeValve.invoke(CompositeValve.java:62)
                at org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve.invoke(CarbonStuckThreadDetectionValve.java:159)
                at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:1025)
                at org.wso2.carbon.tomcat.ext.valves.CarbonContextCreatorValve.invoke(CarbonContextCreatorValve.java:57)
                at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
                at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:445)
                at org.apache.coyote.http11.AbstractHttp11Processor.process(AbstractHttp11Processor.java:1137)
                at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:637)
                at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1775)
                at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:1734)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java)
                at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
                at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: Invalid access token`

Affected versions - IS-KM 5.3.0

--- a/components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/listener/OAuthTokenValidationDASDataPublisher.java
+++ b/components/org.wso2.carbon.identity.data.publisher.oauth/src/main/java/org/wso2/carbon/identity/data/publisher/oauth/listener/OAuthTokenValidationDASDataPublisher.java
@@ -54,8 +54,18 @@ public class OAuthTokenValidationDASDataPublisher extends AbstractOAuthEventInte
                                       OAuth2TokenValidationResponseDTO oAuth2TokenValidationResponseDTO, Map<String,
             Object> params) throws IdentityOAuth2Exception {
 
-        AccessTokenDO accessTokenDO = OAuth2Util.getAccessTokenDOfromTokenIdentifier(oAuth2TokenValidationRequestDTO
-                .getAccessToken().getIdentifier());
+        AccessTokenDO accessTokenDO;
+        try {
+            accessTokenDO = OAuth2Util.getAccessTokenDOfromTokenIdentifier(oAuth2TokenValidationRequestDTO
+                    .getAccessToken().getIdentifier());
+        } catch (IllegalArgumentException e) {
+            // TODO such erroneous cases should be published through another publisher.
+            // Intentionally catch this RuntimeException to break the flow when the access token is invalid.
+            // There can be a different publisher which can publish that data in that erroneous flow.
+            LOG.error("The access token is invalid. Hence failed to publish data through "
+                    + OAuthDataPublisherConstants.OAUTH_TOKEN_VALIDATION_DAS_DATA_PUBLISHER);
+            return;
+        }
 
         StringBuilder authzScopes = new StringBuilder();
         List<String> grantedScopes = Arrays.asList(oAuth2TokenValidationResponseDTO.getScope());

